components/provisioner/e2e_test/test.sh

```diff
@@ -0,0 +1,123 @@
+
+docker network create psql
+docker run --name some-postgres --network psql -e POSTGRES_PASSWORD=somepass -p 5432:5432 --rm -d postgres
+
+sleep 10s
+
+
+export APP_DIRECTOR_URL=https://compass-gateway-auth-oauth.mps.dev.kyma.cloud.sap/director/graphql
+if [[ -z "$APP_DIRECTOR_OAUTH_PATH" ]]; then
+    export APP_DIRECTOR_OAUTH_PATH=/compass-director-secret/secret.yaml
+fi
+
+if ! [[ -f "$APP_DIRECTOR_OAUTH_PATH" ]]; then
+    echo "APP_DIRECTOR_OAUTH_PATH is not set or file doesn't exist $APP_DIRECTOR_OAUTH_PATH"
+    exit 1
+fi
+
+export APP_GARDENER_KUBECONFIG_PATH=${APP_GARDENER_KUBECONFIG_PATH:-$GARDENER_KYMA_PROW_KUBECONFIG}
+export APP_GARDENER_PROJECT=${GARDENER_KYMA_PROW_PROJECT_NAME:-$APP_GARDENER_PROJECT}
+
+ensure_exists APP_GARDENER_KUBECONFIG_PATH
+ensure_exists APP_GARDENER_PROJECT
+
+export GARDENER_PROVIDER=$GARDENER_PROVIDER
+export GARDENER_SECRET_NAME=${GARDENER_KYMA_PROW_PROVIDER_SECRET_NAME:-$GARDENER_SECRET_NAME}
+
+ensure_exists GARDENER_PROVIDER
+ensure_exists GARDENER_SECRET_NAME
+
+export APP_DATABASE_HOST=localhost
+export APP_DATABASE_NAME=provisioner
+export APP_DATABASE_PASSWORD=somepass
+export APP_DATABASE_PORT=5432
+# 👇 this "key leak" is fine, the database is only accesible on localhost and has no persistence
+export APP_DATABASE_SECRET_KEY=f16854073f716495dc933f3cc16de9ee
+export APP_DATABASE_USER=postgres
+
+export APP_PROVISIONING_TIMEOUT_INSTALLATION=90m
+export APP_PROVISIONING_TIMEOUT_UPGRADE=90m
+export APP_PROVISIONING_TIMEOUT_AGENT_CONFIGURATION=90m
+export APP_PROVISIONING_NO_INSTALL_TIMEOUT_AGENT_CONFIGURATION=90m
+export APP_PROVISIONING_TIMEOUT_AGENT_CONNECTION=90m
+export APP_PROVISIONING_TIMEOUT_CLUSTER_CREATION=90m
+export APP_PROVISIONING_NO_INSTALL_TIMEOUT_CLUSTER_CREATION=90m
+export APP_PROVISIONING_TIMEOUT_UPGRADE_TRIGGERING=90m
+
+printf '\n########## SETTING UP THE DB ##########\n\n'
+go run ./pgsetup.go
+
+printf '\n########## BUILDING SCHEMA-MIGRATOR ##########\n\n'
+docker build -t schema-migrator ../../schema-migrator
+
+mkdir migrations
+cp ../../schema-migrator/migrations/provisioner/* ./migrations/
+cp ../../../resources/kcp/charts/provisioner/migrations/* ./migrations/
+
+printf '\n########## MIGRATING THE DB ##########\n\n'
+docker run -v $PWD/migrations:/migrate/migrations/provisioner:ro \
+    --network psql \
+    -e DB_HOST="some-postgres" \
+    -e DB_NAME=$APP_DATABASE_NAME \
+    -e DB_PORT=$APP_DATABASE_PORT \
+    -e DB_USER=$APP_DATABASE_USER \
+    -e DB_PASSWORD=$APP_DATABASE_PASSWORD \
+    -e MIGRATION_PATH=provisioner \
+    -e DIRECTION=up \
+    -e DB_SSL=disable \
+    schema-migrator
+
+printf '\n########## SETTING UP PROVISIONER ##########\n\n'
+go mod download
+go run ../cmd/ | tee "${LOG_DIR}/provisioner.log" &
+export PROVISIONER_PID=$!
+
+sleep 1m
+
+date +"%H:%M:%S"
+
+printf '\n########## RUNNING TESTS ##########\n\n'
+go test -timeout 100m -v ./ | tee "${LOG_DIR}/test.log"
+export TEST_CODE=$?
+
+cleanup
+
+echo exiting with $TEST_CODE
+exit $TEST_CODE
```

Comment:
```
########## SETTING UP PROVISIONER ##########

usage: sleep seconds
tee: /var/log/provisioner.log: Permission denied
15:35:28

########## RUNNING TESTS ##########

tee: /var/log/test.log: Permission denied
=== RUN   TestName
    main_test.go:149: Provisioning a gcp cluster - pts946529
    main_test.go:151:
                Error Trace:    (...)/control-plane/components/provisioner/e2e_test/main_test.go:151
                Error:          Received unexpected error:
                                Post "http://localhost:3000/graphql": dial tcp [::1]:3000: connect: connection refused
                Test:           TestName
    main_test.go:152: {{    }}
    main_test.go:156: Waiting for pts946529 to provision

    main_test.go:158:
                Error Trace:    (...)/control-plane/components/provisioner/e2e_test/main_test.go:158
                Error:          Received unexpected error:
                                Post "http://localhost:3000/graphql": dial tcp [::1]:3000: connect: connection refused
                Test:           TestName
    main_test.go:159: {{{{0}} {  {}}}}
    main_test.go:161: Deprovisioning cluster pts946529
    main_test.go:163:
                Error Trace:   (...)/control-plane/components/provisioner/e2e_test/main_test.go:163
                Error:          Received unexpected error:
                                Post "http://localhost:3000/graphql": dial tcp [::1]:3000: connect: connection refused
                Test:           TestName
    main_test.go:164: {}
--- FAIL: TestName (0.00s)
FAIL
FAIL    github.com/kyma-project/control-plane/components/provisioner/e2e_test   0.101s
FAIL
signal: broken pipe
./test.sh: line 19:  4980 Exit 1                  go run ../cmd/
      4981 Terminated: 15          | tee "${LOG_DIR}/provisioner.log"
exiting with 1
```

Sharing the last outcome I've managed to get. Most important parts:
1. `tee: /var/log/provisioner.log: Permission denied` 
2. `dial tcp [::1]:3000: connect: connection refused`
I'm suspecting that the sleep command is somehow not working as it should but didn't manage to dig into that

Reply:
ad1. this is fine imo, this produces an artifact in pipelines and does nothing locally
ad2 this is caused by broken sleeps, for some reason in pipelines `sleep 10s` works, when locally I cannot provide a unit, so `sleep 10` is correct. That's fixed

Reply:
also, if you want to avoid the first error, you could set a different `LOG_DIR`